### PR TITLE
Add startup flag to specify functions that should be favorably selected by fuzzer

### PR DESCRIPTION
**Pull request by bikramSingh91**
This adds a startup flag "assign_function_tickets" to the fuzzer that takes
a comma separated list of function names and their tickets in the format
<function_name>=<tickets>. Every ticket represents an opportunity for
a function to be chosen from a pool of candidates. By default, every
function has one ticket, and the likelihood of a function being picked
can be increased by allotting it more tickets. Note that in practice,
increasing the number of tickets does not proportionally increase the
likelihood of selection, as the selection process involves filtering
the pool of candidates by a required return type so not all functions
may compete against the same number of functions at every instance.
Number of tickets must be a positive integer.
Example:
--assign_function_tickets "eq=3,floor=5"

Test Plan:

Ran fuzzer for 60 secs with:

Case 1:
--assign_function_tickets "reverse=1"

functionName numTimesSelected proportionOfTimesSelected numProcessedRows
subscript 4698 6.24% 50935
array_min 3039 4.04% 31115
array_max 3027 4.02% 34004
least 1830 2.43% 10932
greatest 1749 2.32% 9115
clamp 1492 1.98% 13815
floor 1416 1.88% 20626
divide 1404 1.87% 18279
negate 1399 1.86% 19363
sign 1367 1.82% 16842
multiply 1362 1.81% 19842
ceil 1346 1.79% 16744
minus 1341 1.78% 17084
mod 1335 1.77% 18587
abs 1324 1.76% 17189
round 1315 1.75% 17258
ceiling 1310 1.74% 19120
plus 1299 1.73% 18900
reverse 1100 1.46% 11350

Case 2:
--assign_function_tickets "reverse=6"

subscript 4914 6.17% 56986
reverse 3209 4.03% 35340
array_max 3180 3.99% 38363
array_min 3171 3.98% 35746
greatest 1787 2.24% 9678
least 1763 2.21% 10213
multiply 1533 1.93% 24829
ceil 1518 1.91% 21930
round 1517 1.91% 19357
abs 1499 1.88% 20086
plus 1489 1.87% 19672
mod 1487 1.87% 19735
minus 1466 1.84% 19829
ceiling 1443 1.81% 22217
negate 1438 1.81% 21344
clamp 1433 1.80% 14652
sign 1432 1.80% 20460
floor 1423 1.79% 18669
divide 1408 1.77% 19904
date_add 1062 1.33% 4755
date_trunc 1021 1.28% 8358
slice 777 0.98% 7429

### CODE REVIEW

**Comment by bikramSingh91** [comment]
This is a follow-up PR after #3695 that leverages it to enable this feature.
Fixes #3388

**Comment by bikramSingh91** [comment]
@kgpai @kagamiori Can you please take a look when you get a chance? Thanks!

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 1]
```diff
@@ -447,10 +514,14 @@ ExpressionFuzzer::ExpressionFuzzer(
...
         typeToExpressionList_[returnType].back() != it.name) {
-      // Ensure only one entry for a function name is added. This
+      // Ensure entries for a function name are added only once. This
       // gives all others a fair chance to be selected. Since signatures
       // are sorted on the function name this check will always work.
-      typeToExpressionList_[returnType].push_back(it.name);
+      int tickets = getTickets(it.name);
+      // Add multiple entries to increase likelihood of its selection.
+      for (int i = 0; i < tickets; i++) {
+        typeToExpressionList_[returnType].push_back(it.name);
```
Comment (kagamiori) [inline velox/expression/tests/ExpressionFuzzer.cpp:523]: nit: Not a problem with the current code, but `typeToExpressionList_` actually maps the return type name to function names that could return this type. Should we rename it to something like `typeToFunctionName`?

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 2]
```diff
@@ -447,10 +514,14 @@ ExpressionFuzzer::ExpressionFuzzer(
...
+      }
     }
     expressionToSignature_[it.name][returnType].push_back(&it);
```
Comment (kagamiori) [inline velox/expression/tests/ExpressionFuzzer.cpp:526]: nit: Not a problem with the current code, but `expressionToSignature_` maps function name to its return type names and further to the callable signatures of the function and return type. On the other hand, the comment in ExpressionFuzzer.h says "Maps the base name of a *concrete* return type signature to the function name...". Should we update the comment to be consistent?

**Review by kagamiori — APPROVED** [review]
Looks good to me. Thank you for adding this handy feature to allow controlling the function selection in expression generation!

File: velox/expression/tests/ExpressionFuzzer.cpp (hunk 1 above)
Comment (bikramSingh91) [reply velox/expression/tests/ExpressionFuzzer.cpp:523]: Thanks for the nits, I'll address these in a follow up refactor

File: velox/expression/tests/ExpressionFuzzer.cpp (hunk 2 above)
Comment (bikramSingh91) [reply velox/expression/tests/ExpressionFuzzer.cpp:526]: Thanks for the nits, I'll address these in a follow up refactor

**Pull request closed without merging**